lib/deploy_complexity/changed_files.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Parses git shortstat output
+# `git diff --name-only v0.4.0...v0.5.0`
+class ChangedFiles
+  def initialize(names, versioned_url)
+    @names = names.split(/\n/)
+    @versioned_url = versioned_url
+  end
+
+  def migrations
+    @names.grep(%r{^db/migrate}).map do |line|
+      @versioned_url + line.chomp
+    end
+  end
+
+  def elm_packages
+    @names.grep(%r{(^|/)elm\.json$}).map(&:chomp)
+  end
+
+  def ruby_dependencies
```

Comment:
Right now these all output lists, even though there will only be one Gemfile.lock and one package-lock.json - would there ever be a project where these are not at the top level?